corehq/apps/repeaters/tasks.py

```diff
@@ -60,9 +61,10 @@ def process_repeat_record(repeat_record):
         if repeat_record.repeater.doc_type.endswith(DELETED_SUFFIX):
             if not repeat_record.doc_type.endswith(DELETED_SUFFIX):
                 repeat_record.doc_type += DELETED_SUFFIX
+                repeat_record.save()
         else:
-            repeat_record.fire()
-        repeat_record.save()
+            if repeat_record.state == RECORD_PENDING_STATE or repeat_record.state == RECORD_FAILURE_STATE:
```

Comment:
seems right not to send when it's been cancelled, i think the cancelled state was added after this logic was added, so makes sense that this doesn't account for it.

nit: can make this a `elif` instead of an `else` then an `if`